### ISSUE TITLE
Setting accessibilityLabel on <View> doesn't work without also setting accessibilityRole

### ISSUE THREAD

**Issue by miczhongMSFT** [microsoft/react-native-windows#3836]
### Environment
Xbox Events Beta App

1. `react-native -v`:
2. `npm ls rnpm-plugin-windows`: rnpm-plugin-windows@0.2.13 
3. `npm ls react-native-windows`: react-native-windows@0.60.0-vnext-headlessjs.49 
4. `node -v`: 10.15.0
5. `npm -v`: 6.4.1
6. `yarn --version`:

Then, specify:

- Target Platform: UWP

- Target Platform Version(s):

- Target Device(s): All Xbox 

- Development Operating System: Xbox

- Visual Studio Version: 

- Build Configuration:

### Steps to Reproduce

The accessibility label of a parent `<View>` is not being read. This works perfectly on a web solution, but not in any XAML solutions (Xbox/Windows).

Let's say we are making a section that is filled with buttons, and we'd like to give this section a header called "Related buttons section:
![image](https://user-images.githubusercontent.com/46689071/72113757-42116000-32f6-11ea-8e59-192c95261749.png)
The react native code would look something like this:
`<View accessibilityLabel={"Related Buttons Section"}>`
    `<Text>"Related Buttons Section"</ Text>`
    `<Button title={"button 1"}/>`
    `<Button title={"button 2"}/>`
`</ View>`

When we focus on the first button, narrator should read out "Related Buttons Section. Button 1, button". However, the `accessibilityLabel` attached to `<View>` is completely being ignored. This is an issue for visually-less-able, because they will not know what the section is about.

Here is a [web version sample code](https://snack.expo.io/H19gHEHgL?session_id=snack-session-YAI9qDd__&preview=true&platform=web&iframeId=f0rtty9di5&supportedPlatforms=ios,android,web&name=SectionList&description=Example%20usage&waitForData=true). Feel free to turn on windows narrator and checkout narrator's behavior.

(Write your steps here:)
Note: 
1.  Create a "section" like the code above as a Xbox Application
2. Turn on Xbox's narrator
3. Bring focus to the first button in the section

### Expected Behavior
When we focus on the first button, narrator should read out "Related Buttons Section. Button 1, button". 

### Actual Behavior
Narrator only reads "button". 
Accessibility label on the outer <View> is completely ignored. Even the button's `title` is not being read, unless an accessibility label is explicitly assigned to the "Button". 

(Write what happened. Add screenshots!)

### Reproducible Demo

[Working sample on web](https://snack.expo.io/H19gHEHgL?session_id=snack-session-YAI9qDd__&preview=true&platform=web&iframeId=f0rtty9di5&supportedPlatforms=ios,android,web&name=SectionList&description=Example%20usage&waitForData=true)

(Paste the link to an example project and exact instructions to reproduce the issue.)

**Comment by miczhongMSFT** [comment]
This is seems to be a result of <Text> component not respecting `accessibilityRole` of "header"

**Comment by jonthysell** [comment]
This is an issue because while you can set the `accessibilityLabel`, and it gets mapped to `AutomationProperties.Name`, but without an `acessibilityRole` then we have a vanilla `ViewPanel` without an `AutomationPeer`, so that name never gets exposed to UIA.

If you add an `accessibilityRole`, and thereby get the wrapper ViewControl, then it will appear properly in UIA, but now the View is focusable, messsing up the desired navigation experience.

I added `Needs WinUI 3` because one of the biggest RNW requirements is to get the access we need (through WinUI 3) to build a proper View XAML control that supports all of the features and flexibility of RN Views.

**Comment by jonthysell** [comment]
The work for WinUI 3: #3111

**Comment by kmelmon** [comment]
@jonthysell is it possible for us to fix this without WinUI3.0?  It seems like a basic scenario we need to support

**Comment by jonthysell** [comment]
> 
> 
> @jonthysell is it possible for us to fix this without WinUI3.0? It seems like a basic scenario we need to support

To do this we need to get rid of the whole ViewControl>Border>ViewPanel swapping madness, because as long as we maintain that, we need to be able to delete/invalidate automation peers, which is not exposed to end developers. Or we can save off properties/children, remove the panel from the tree, create a new panel, and re-populate it, and have flicker.

We chose to only put the peer on ViewControl, to meet Calendar's requirements, so the act of adding/removing the ViewControl effectively removes the old peer.

The right thing to do. is either we build a proper ReactView control in WinUI3 without all of this swapping, or we expose the delete peer APIs, which again would require WinUI3.

Right now Xbox has an easy app-side workaround for the scenario, so I'm inclined to wait for WinUI3.

**Comment by jonthysell** [comment]
Updating the title to reflect the actual issue

**Comment by chrisglein** [comment]
Okay, sounds like there is a workaround, and that fixing this without changes to XAML (via WinUI3+) would be maybe more complicated than it's worth. Since this is a fundamental accessibility issue let's evaluate what we could do without WinUI3 next milestone.

**Comment by asklar** [comment]
@jonthysell @chrisglein  can you describe what we need from winui 3? we should file a bug linked to this one to track when we can implement using whatever winui 3.x gives us in this respect.

**Comment by chrisglein** [comment]
> @jonthysell @chrisglein can you describe what we need from winui 3? we should file a bug linked to this one to track when we can implement using whatever winui 3.x gives us in this respect.

Jon has some issues on this repo (most notable #3875), but it seems like we still need one opened on the WinUI repo. At least as it relates to automation peers there is [this](https://github.com/microsoft/microsoft-ui-xaml/issues/1349).

**Comment by chiaramooney** [comment]
Now working on new architecture. Closing this bug; reopen if issue on new arch.
